lib/api/src/index.tsx

```diff
@@ -0,0 +1,267 @@
+import React, { Component } from 'react';
+import memoize from 'memoizerific';
+
+import Events from '@storybook/core-events';
+import { Collection, Types } from '@storybook/addons';
+import initProviderApi from './init-provider-api';
+
+import { createContext } from './context';
+import Store from './store';
+import getInitialState from './initial-state';
+
+import initAddons from './modules/addons';
```

Comment:
I think the modules should remain in `lib/ui`

Reply:
At least the modules that are directly relevant to the UI, such as the layout state.

Reply:
possibly, do you have a suggestion on how to hook that up?

I feel like there a pros and cons to both approaches. Having the global state in 1 location is not a bad thing?

Reply:
I guess it depends what you mean. I would expect the state to all be stored in the one "store" but the code for setting/changing the UI's state would live in `lib/ui`. Similarly, addons set their own state in the same store but their store setting state lives in their codebase.

I guess the question is how to make the APIs that the UI creates available to the addons in a principled/typed way. Would it make sense that if an addon wants to alter the UI's state (e.g. set fullscreen) that it should import something from `@storybook/ui`? I feel like that would be sensible.